### PR TITLE
Fix #109: Basic Testing Infrastructure to Provide Easy to use Setup and Helper functions

### LINKED ISSUE #109: test: Basic Testing Infrastructure to Provide Easy to use Setup and Helper functions
### Objective
Making a universal setup function that help to manage state of the testing repository and provide easy to access functions for testing.

### Detailed Implementation Steps
1. Making a universal setup function to use in integration testing and managing state of the repository.
2. Making helper functions that represent different conditions for testing.

### Files to Modify
- Add `test/test_setup.go`
- Add `test/test_helpers.go`

### Acceptance Criteria
- [x] Command `kitkat ...` works as expected.
- [x] `kitkat log` / `status` shows correct output.
- [x] Edge cases (like missing files) are handled.

### Track Selection
- [x] Track 3: Systems Engineering

### PR DESCRIPTION
**Pull request by tuxy-streamer**
# Description
Fixes #109

# Implementation Details
- Added test_setup.go
- Added test_handlers.go

# Verification (Mandatory)

# Track Selection
- [x] Track 3: Intermediate Code

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own changes
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have run `go fmt ./...` locally (not applicable – documentation only)
- [x] I have verified all "Acceptance Criteria" listed in the issue

### CODE REVIEW

**Comment by Atharv3221** [comment]
We can use it in go too

**Comment by Atharv3221** [comment]
Okay this is standard go way

**Comment by Atharv3221** [comment]
@tuxy-streamer I feel it bad creating temporary environment for testing

**Comment by tuxy-streamer** [comment]
@Atharv3221 if you're talking about the path in the assertion functions then those are the core paths to be checked.
I don't think it will be a problem because it shouldn't change for anyone.

File: test/test_setup.go [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+	}
+}
+
+func logError(t *testing.T, msg string, err error) {
+	if err != nil {
+		t.Errorf("%s: %v", msg, err)
+	}
+}
+
+// setupSandbox creates an isolated temporary environment and optionally initializes a repo.
```
Comment (Atharv3221) [inline test/test_setup.go]: why creating temporary environment ?
